### ISSUE TITLE
[CascadingCallWrapping] Doesn't report the point of the failure

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#5043]
## Expected Behavior
<img width="896" alt="Captura de pantalla 2022-07-06 a las 19 06 43" src="https://user-images.githubusercontent.com/721244/177605576-b57b0e95-b465-4754-93d1-1c305ac45c37.png">

## Observed Behavior
<img width="892" alt="Captura de pantalla 2022-07-06 a las 19 06 56" src="https://user-images.githubusercontent.com/721244/177605606-e9dca6ea-461c-447f-b6bf-dc5e086845af.png">

Two images worht 1000 words. I think that this rule should flag the code that is wrong instead of all the chain.

## Context
There are multiple reasons for this:
- I have really long chains on my project and the html report only shows the first 4 lines of the code that's wrong. The problem is that the error is below those 4 lines so the snippet is useless
- I usualy use the console log I check the line and go there. But what the console log say is this: `path/File.kt:128:25` but the error is not in the line 128. That's the start of the chain.

## Your Environment
* Version of detekt used: 1.21.0-RC2

**Comment by dimsuz** [comment]
I'd like to work on this issue if nobody is already fixing it.

**Comment by dimsuz** [comment]
Ah, I still misread that PR, turns out it's for another issue. I'll tag this one then! :)

**Comment by dimsuz** [comment]
I have almost done it, cant wrap my head on how to construct an `Entity` in error report for one particular case with elvis.

I want do highlight it like this:

```kotlin
val a = 0
   .plus(0) ?: 0
            ~~~~
```

In this case this is a `KtBinaryExpression` and I somehow need to join `expression.operationReference` and `expression.right` and report their "cumulative" location. 

If only I could do something like `Entity.from(listOf(expr1, expr2))`, but I guess that such an API wouldn't be well-defined for **any** expressions, only a sibling ones...

Perhaps I need to somehow mess with `Location`s and/or `startOffset` and `endOffset`, but there are so many of them in the Psi API and zero documentation. If someone has an immediate advice, I'm all ears, otherwise I'll debug a bit more.

**Comment by dimsuz** [comment]
>  If someone has an immediate advice, I'm all ears, otherwise I'll debug a bit more.

Figured this out, see the PR.

**Comment by BraisGabin** [comment]
Thanks for the fix :)
